### DISCUSSION TITLE
How to open web-site (URL)?

### DISCUSSION THREAD

**Discussion by code-by** [telegraf/telegraf#1745] [Q&A] [answered]
Hello
I'd like to open web-site (URL) ((in-place (like send link to somebody) or open in browser)) by clicking button made by `keyboard` (not `inlineKeyboard`). Which methods I could use or there only possibility to `hear` button clicks and output URL in reply?
thank you

**Comment by MKRhere** [comment, marked as answer]
As you can see here:  https://core.telegram.org/bots/api#keyboardbutton
KeyboardButton does not allow having a URL button. Only inline keyboard does.

**Reply by code-by** [reply]
So only one of the possibility to response with URL to user is `hear` from `KeyboardButtton` and reply with URL?

**Reply by MKRhere** [reply]
That's one way, you could also just use an inline button.

**Reply by code-by** [reply]
> That's one way, you could also just use an inline button.

I see, thank you. I'd like to use `keyboard` for user interaction instead of inline buttons. Btw seems it is not possible to use both kind of keyboards simultaneously...
